### ISSUE TITLE
Pyspark models fail to import 

### ISSUE THREAD

**Issue by tylerfordML** [Hydrospheredata/spark-ml-serving#28]
Any pyspark models with data contained in them fail to import. The exported data parameters are exported as an array and not a list. I am assuming this is an issue between scala and python. Logistic regression and  random forest so far have issues. It looks like the data util class is looking for a list of impurityStats. Pyspark exports this out as an array and not a list. Logistic regression fails on colPtrs in spark 2.2 classification

**Comment by KineticCookie** [comment]
Hello @tylerfordML 
Our focus was on Spark ML models, and we assumed that PySpark exports models like Spark. It's strange though, that they have different export result. Nevertheless, I'll investigate this issue.
Do you have, by chance, any PySpark model to reproduce this errors?

**Comment by tylerfordML** [comment]
Yes, I will send over some models.

**Comment by tylerfordML** [comment]
On Mar 21, 2018 7:04 AM, "Bulat" <notifications@github.com> wrote:

Hello @tylerfordML <https://github.com/tylerfordml>
Our focus was on Spark ML models, and we assumed that PySpark exports
models like Spark. It's strange though, that they have different export
result. Nevertheless, I'll investigate this issue.
Do you have, by chance, any PySpark model to reproduce this errors?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Hydrospheredata/spark-ml-serving/issues/28#issuecomment-374929142>,
or mute the thread

**Comment by tylerfordML** [comment]
also, I was doing some debugging and it looks like the pyspark model is
using a dense matrix for logistic regression. here is the meta data for
both the logistic regression demo model and the pyspark one that i
exported. looks like  *coefficientMatrix *are exported out differently

*Pyspark*
Row(numClasses=2, numFeatures=1000, interceptVector=DenseVector([-2.6407]),
*coefficientMatrix=SparseMatrix*(1, 1000, [0, 11], [94, 105, 170, 181, 248,
282, 361, 417, 722, 878, 953], [-0.6361, 2.4778, 1.8943, -1.1022, 1.4095,
1.4095, -0.6361, 1.4095, 1.8943, 1.8943, -1.1022], True),
isMultinomial=False)


Row(numClasses=169, numFeatures=10000, interceptVector=DenseVector([5.8549,
5.528, 6.3204, 5.16, 4.3923, 3.9984, 3.9261, 4.1504, 3.7516, 3.7137,
3.5113, 3.5234, 3.5497, 3.313, 3.3835, 3.3889, 3.3108, 3.3398, 3.188,
3.1332, 3.1509, 3.1544, 3.1195, 3.0828, 3.1164, 3.0543, 2.9847, 3.0095,
2.9282, 2.7924, 2.7549, 2.6805, 2.6634, 2.6704, 2.6102, 2.6006, 2.5927,
2.49, 2.4093, 2.2619, 2.2315, 2.1623, 2.0759, 2.1153, 1.9827, 1.9621,
1.9433, 1.9199, 1.8472, 1.7833, 1.8227, 1.7746, 1.7802, 1.6642, 1.5703,
1.602, 1.594, 1.5616, 1.4989, 1.3277, 1.2852, 1.1603, 1.0288, 0.7947,
0.8141, 0.7237, 0.7792, 0.6426, 0.5509, 0.5387, 0.5833, 0.4568, 0.4929,
0.367, 0.3118, 0.381, 0.2235, 0.2835, 0.281, 0.2387, 0.2084, 0.037,
-0.0201, 0.0561, 0.1092, -0.0811, -0.3152, -0.2149, -0.3977, -0.3977,
-0.6579, -0.5862, -0.4876, -0.9585, -0.863, -0.735, -0.5523, -0.8185,
-0.863, -1.0099, -1.0099, -0.9585, -1.2466, -1.2466, -1.5571, -1.5571,
-1.5571, -1.3899, -1.3157, -1.8758, -1.8758, -1.6525, -1.6525, -1.7579,
-1.7579, -1.8758, -1.8758, -2.0094, -2.1636, -2.0094, -2.0094, -2.1636,
-2.346, -2.857, -2.346, -3.2626, -2.1636, -2.3461, -2.3461, -2.346,
-2.5693, -2.857, -2.346, -2.857, -2.857, -3.2626, -3.2626, -2.857, -2.857,
-2.857, -2.857, -3.2626, -2.857, -2.857, -3.2626, -3.2626, -3.2626,
-3.9558, -3.9558, -3.2626, -3.2626, -3.2626, -3.2626, -3.2626, -3.9558,
-3.2626, -3.2626, -3.2626, -3.9558, -3.2626, -3.9558, -3.9558, -3.2626,
-3.2626, -3.2626, -3.2626, -3.2626, -0.0135, -0.0135]),
*coefficientMatrix=DenseMatrix(*169, 10000, [0.0054, 0.0128, 0.0118,
0.0321, 0.0127, -0.0143, 0.0102, -0.0093, ..., -0.0007, -0.0004, -0.0002,
-0.0008, -0.0005, -0.0, -0.0012, -0.0009], True), isMultinomial=True)

**Comment by KineticCookie** [comment]
So, I investigated the behaviour behind LogisticRegression. Seems like I implemented matrices load with some incorrect assumptions. (Specifically, I used to create matrices using built-in factory methods, and seems like they have problems with creating transposed matrices).
 I made fixes in separate branch and wrote test for loading a PySpark LogisticRegression model.

As soon as I finish with random forests, I will merge and release fixes.
